cmd/fetch.go

```diff
@@ -116,13 +122,28 @@ func Query() *cli.Command {
 
 				queryStr = addLimitToQuery(queryStr, c.Int64("limit"), conn, parser, dialect)
 			}
-			if querier, ok := conn.(interface {
+			if querier, ok := conn.(interface { //nolint: nestif
 				SelectWithSchema(ctx context.Context, q *query.Query) (*query.QueryResult, error)
 			}); ok {
-				ctx := context.Background()
+				// Create context with timeout if specified
+				var ctx context.Context
+				var cancel context.CancelFunc
+				if timeout := c.Int("timeout"); timeout > 0 {
+					ctx, cancel = context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
+					defer cancel()
+				} else {
+					ctx = context.Background()
+				}
```

Comment:
meh, this would work only if the CLI is still running. how do we attach this information to the underlying database queries as well, e.g. directly to the snowflake job?